File: solo-leveling-ui-service/src/main/java/com/sleepkqq/sololeveling/ui/broadcast/TaskUpdatesTopic.java
package com.sleepkqq.sololeveling.ui.broadcast;

import com.sleepkqq.sololeveling.avro.notification.Notification;
import java.util.function.Consumer;
import org.springframework.stereotype.Component;

@Component
public class TaskUpdatesTopic extends AbstractTopic<Consumer<Notification>> {

  @Override
  protected void notifyListener(Consumer<Notification> listener, Notification notification) {
    listener.accept(notification);
  }
}

File: solo-leveling-ui-service/src/main/kotlin/com.sleepkqq.sololeveling/ui/broadcast/TaskUpdatesTopic.kt
package com.sleepkqq.sololeveling.ui.broadcast

import com.sleepkqq.sololeveling.avro.notification.Notification
import org.springframework.stereotype.Component
import java.util.function.Consumer

@Component
class TaskUpdatesTopic : AbstractTopic<Consumer<Notification>>() {

	override fun notifyListener(listener: Consumer<Notification>, notification: Notification) =
		listener.accept(notification)
}
